src/cli/serve/serve.js

```diff
@@ -223,6 +223,7 @@ export default function (program) {
           repl: !!opts.repl,
           basePath: !!opts.basePath,
           optimize: !!opts.optimize,
+          pluginPath: opts.pluginPath,
```

Comment:
__suggestion:__ hmm, I guess you get this already as a `plugins.paths` config value? If I'm not mistaken, you just need two things:
* add `paths: schema.arrayOf(schema.string(), { defaultValue: [] }),` to [`PluginsConfig` schema](https://github.com/elastic/kibana/blob/master/src/core/server/plugins/plugins_config.ts#L23-L25)
* add expose this value in [`LegacyObjectToConfigAdapter`](https://github.com/elastic/kibana/blob/master/src/core/server/legacy/config/legacy_object_to_config_adapter.ts#L75-L79)

That way you won't need to do any changes here or in Env...

Reply:
Great, this makes more sense to me. The delineation between properties of `Env`, `Env#cliArgs` and `Env#configs` is not apparent to me. I would expect that all ways of configuring (environment variables, CLI switches, yaml files) would be merged/resolved into a single config object. This way the consuming code doesn't need to care where the config value came from.

What's the reasoning for exposing the `cliArgs` object directly?

Reply:
> I would expect that all ways of configuring (environment variables, CLI switches, yaml files) would be merged/resolved into a single config object. 

Definitely agree

> What's the reasoning for exposing the cliArgs object directly?

IIRC, it's just for legacy compatibility and used only in `LegacyService` to pass that to `ClusterManager` (that we should eventually get rid of, or at least remove from main code) and to determine whether we need repl (can just become property on env or something like this).

Reply:
Got it, thanks for the context here!